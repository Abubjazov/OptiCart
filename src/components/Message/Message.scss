@import '../../variables';

.message {
	h2 {
		color: $main-color;
		font-size: 90px;
		font-weight: 900;
		margin-bottom: 10px;
	}

	.link {
		display: block;
		margin-top: 70px;
		color: $black;
		text-decoration: none;
		transition: all 0.1s linear;
		font-size: 70px;
		font-style: normal;
		vertical-align: middle;

		span {
			color: $main-color;
		}

		&:hover {
			color: $main-color;

			span {
				color: $black;
			}
		}
	}
}
