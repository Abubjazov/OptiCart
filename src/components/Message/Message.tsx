import './Message.scss'

const defaultMessage = 'There are no sneakers in your cart yet!!! )'
const defaultLinkText = 'Choose*YOURS!'

export const Message = ({
	message = defaultMessage,
	linkText = defaultLinkText,
}: MessageProps): JSX.Element => (
	<div role='alert' className='message'>
		<h2 tabIndex={0}>{message}</h2>

		<a href='/optiCart/'>
			{linkText?.split('*')[0]} <span>{linkText?.split('*')[1]}</span>
		</a>
	</div>
)

export interface MessageProps {
	message?: string | null
	linkText?: string | null
}
