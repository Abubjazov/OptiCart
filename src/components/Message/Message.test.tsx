import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'

import { Message } from './Message'

describe('Component: Header', () => {
	test('should render Header', () => {
		const { asFragment } = render(
			<BrowserRouter>
				<Message />
			</BrowserRouter>
		)

		expect(asFragment()).toMatchSnapshot()
	})
})
