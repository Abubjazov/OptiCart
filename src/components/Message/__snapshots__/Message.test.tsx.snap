// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Message should render default Message 1`] = `
<DocumentFragment>
  <div
    class="message"
    role="alert"
  >
    <h2
      tabindex="0"
    >
      There are no sneakers in your cart yet!!! )
    </h2>
    <a
      href="/optiCart/"
    >
      Choose 
      <span>
        YOURS!
      </span>
    </a>
  </div>
</DocumentFragment>
`;

exports[`Component: Message should render modified Message 1`] = `
<DocumentFragment>
  <div
    class="message"
    role="alert"
  >
    <h2
      tabindex="0"
    >
      Test message
    </h2>
    <a
      href="/optiCart/"
    >
      Test 
      <span>
        Link text
      </span>
    </a>
  </div>
</DocumentFragment>
`;
